src/Foundation/NSUrl.cs

```diff
@@ -30,6 +30,32 @@ public NSUrl (string path, string relativeToUrl)
 		{
 		}
 
+		public NSUrl (string urlString)
+		{
+			Handle = _Init (urlString);
+			if (Handle == IntPtr.Zero)
+				throw new ArgumentException ("Malformed url");
```

Comment:
Oh, the fact that we're throwing "Could not initialize an instance of the type" was the source of the bug. See other comment.